.github/problem-matchers/dtc.json

```diff
@@ -0,0 +1,19 @@
+{
+    "problemMatcher": [
+        {
+            "__comment_owner": "match dtc parsing errors",
+            "owner": "dtc",
+            "pattern": [
+                {
+                    "__comment_regexp1": "Error: arch/arm64/boot/dts/qcom/sc7280-idp.dts:27.1-9 Label or path apps_rs not found",
+                    "regexp": "^(Error):\\s+([a-z|A-Z|0-9|\\/|.|\\-|_\\+]+):([0-9|\\-]+).([0-9|\\-]+)\\s+(.*)$",
```

Comment:
When I wrote the DTS error parser before, I noted its very unusual line/col output style. I don't think what you have here is correct. I think it spits out ${start_line}.${start_col}-[${end_line}.]${end_col}. So, I think you want:

...`:([0-9]+)\\.([0-9]+)-[0-9]+(?:\\.[0-9]+)\\s+(.*)$"`

(Note that the "." matches need to be `\\.` if you want to match a literal ".")

Reply:
> I think it spits out ${start_line}.${start_col}-[${end_line}.]${end_col}.

Almost. It is either

* `${start_line}.${start_col}`
* `${start_line}.${start_col}-${end_col}`
* `${start_line}.${start_col}-${end_line}.${end_col}`

if I am reading the [dtc sources](https://git.kernel.org/pub/scm/utils/dtc/dtc.git/tree/srcpos.c#n296) correctly.

My current pattern should work for matching those BUT it does not getting the grouping correct, which appears hard to do because of how different the third form is from the first two. I think it is probably just work removing the grouping altogether and changing it to

`...:[0-9|\\.|\\-]+\\s+(.*)$`

Thoughts?

Reply:
I think this would work? `"^(Error):\\s+([^:]+):([0-9]+)\\.([0-9]+)(?:[-0-9\\.]+)?\s+(.*)$"`

Reply:
Yes, that gets us close enough. I don't really think that we care that much about `dtc` errors given that they are never as a result of clang.